include/lbann/layers/learning/fully_connected.hpp

```diff
@@ -294,32 +215,33 @@ class fully_connected_layer : public learning_layer {
     }
 
     // Setup weights
-    if (this->get_data_layout() == data_layout::MODEL_PARALLEL) {
+    switch (get_data_layout()) {
```

Comment:
It should be, but I think it would make the code messier. How about we duplicate the weights matrix by default and only use a special distribution for the model-parallel case?